pom.xml

```diff
@@ -693,6 +670,145 @@
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-compiler-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>default-compile</id>
+                                <phase>none</phase>
+                            </execution>
+                            <execution>
+                                <id>default-testCompile</id>
+                                <configuration>
+                                    <release>9</release>
+                                    <encoding>UTF-8</encoding>
+                                    <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                    <fork>false</fork>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>java-9-compile</id>
+                                <goals>
+                                    <goal>compile</goal>
+                                </goals>
+                                <configuration>
+                                    <release>9</release>
+                                    <!-- no excludes: compile everything to ensure module-info contains right entries -->
+                                    <encoding>UTF-8</encoding>
+                                    <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                    <fork>false</fork>
+                                    <compilerArgument>-proc:none</compilerArgument>
+                                    <excludes>
+                                        <exclude>software/amazon/awssdk/services/kinesis/model/SubscribeToShardEvent.java</exclude>
+                                        <exclude>software/amazon/awssdk/services/kinesis/KinesisAsyncClient.java</exclude>
+                                        <exclude>software/amazon/awssdk/services/kinesis/DefaultKinesisAsyncClient.java</exclude>
+                                        <exclude>software/amazon/awssdk/services/kinesis/DefaultKinesisBaseClientBuilder.java</exclude>
+                                        <exclude>software/amazon/awssdk/services/sms/model/InternalError.java</exclude>
+                                        <exclude>software/amazon/awssdk/services/kinesis/transform/SubscribeToShardResponseUnmarshaller.java</exclude>
+                                    </excludes>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>java-8-compile</id>
```

Comment:
What do these 2 different executions do? It looks like we compile twice?

Reply:
Yes, that is actually the intent with this change. I added a bit of detail to the description of changes, but Java9+ compilation happens first, then relevant class files are overwritten with java8 class files aside from module-info.class. The resulting JAR works for both JRE8 classpath invocation, and JRE9+ module-path invocation.

Reply:
Interesting... how much does this slow down a full `mvn clean install`?

Reply:
Baseline on m6a.2xlarge was `1h7m`, and after dual compilation it took `1h21m`.